### PR TITLE
feat: view key type wrapper (1/2)

### PR DESCRIPTION
**Pull request by jowparks**
## Summary
Creates a type wrapper in Rust. Working towards storing the viewKey (hex from this PR) in the wallet/account instead of the `incomingViewKey`
## Testing Plan
Tests pass
## Breaking Change

Is this a breaking change? If yes, add notes below on why this is breaking and
what additional work is required, if any.

```
[ ] Yes
```

### CODE REVIEW

File: ironfish-rust/src/keys/mod.rs
Comment (jowparks) [inline ironfish-rust/src/keys/mod.rs:216]: We will delete this method in a subsequent PR and replace with `ViewKey`

File: ironfish-rust/src/keys/mod.rs
Comment (jowparks) [inline ironfish-rust/src/keys/mod.rs:222]: We will delete this method in a subsequent PR and replace with ViewKey

File: ironfish-rust/src/keys/view_keys.rs
Comment (jowparks) [inline ironfish-rust/src/keys/view_keys.rs:114]: Not stoked about these `unwrap`s, but the code will get way uglier if we try to deal with the underlying types of `CtOption`. I was thinking of extending the CtOption in our codebase with something like
```
impl From<CtOption<T>> for Result<T, IronfishError> {
...
```

But we don't include the `subtle` library dependency, so not sure if we can do that without explicitly adding the dependency.

File: ironfish-rust/src/keys/view_keys.rs
Comment (jowparks) [inline ironfish-rust/src/keys/view_keys.rs:197]: Unused struct, deleteing

File: ironfish-rust/src/keys/view_keys.rs
Comment (jowparks) [inline ironfish-rust/src/keys/view_keys.rs:104]: I will put in subsequent PR to rename this error to `InvalidViewKey`, it is present in a few places and has incorrect tense.

**Review by dguenther — APPROVED** [review]
This generally looks good to me. I wonder if we could have simplified some of the renaming, but I don't have any objections here since `ViewKeys` isn't the best name, and we've tended to create our own structs vs extending the zcash ones (`ViewingKey`).

I'd like to wait for @mat-if 's review too before merging

File: ironfish-rust/src/keys/view_keys.rs
Comment (jowparks) [inline ironfish-rust/src/keys/view_keys.rs:98]: It is possible for us to tack on `outgoing_view_key` here. It would also mean we could use a name like `FullViewKey` wonder if the wallet cleanliness is worth the added change? We could also make this change later, which probably is the safest option considering we are close to launch

File: ironfish-rust/src/keys/mod.rs
Comment (mat-if) [inline ironfish-rust/src/keys/mod.rs:216]: Can you clean up this comment? Think it got clobbered when you deleted the other struct

**Review by mat-if — APPROVED** [review]
Looks good - just a few minor nits. 👍

File: ironfish-rust/src/keys/view_keys.rs
Comment (jowparks) [reply ironfish-rust/src/keys/view_keys.rs:104]: Nice, ty for suggestion

File: ironfish-rust/src/keys/test.rs
Comment (jowparks) [reply ironfish-rust/src/keys/test.rs:141]: I was wondering why the `keys` had a separate test file, I just shrugged and added the test there. SGTM, moving

File: ironfish-rust/src/keys/view_keys.rs
Comment (jowparks) [reply ironfish-rust/src/keys/view_keys.rs:118]: Oh nice, I like it. Thanks for the sugar, tastes goooood

File: ironfish-rust/src/keys/view_keys.rs
Comment (jowparks) [reply ironfish-rust/src/keys/view_keys.rs:98]: Closing comment, as well will do this as a followup later

**Pull request merged**
